changelog/changelog.py

```diff
@@ -177,20 +177,24 @@ def sct_changelog_generator(items):
     lines = []
     for item in items:
         title = item['title']
-        sct_labels = [l['name'] for l in item['labels'] if "sct_" in l['name']]
+        sct_labels = sorted(l['name'] for l in item['labels'] if "sct_" in l['name'])
         breaks_compat = any(l['name'] == 'compatibility' for l in item['labels'])
         pr_url = item['html_url']
+        sorting_key = (sct_labels, item['number'])
 
         if breaks_compat:
             compat_msg = "**WARNING: Breaks compatibility with previous version.**"
         else:
             compat_msg = ""
 
         if sct_labels:
-            lines.append(f" - **{', '.join(label for label in sct_labels)}:** {title}. {compat_msg} [View pull request]({pr_url})\n")
+            labels_msg = f"**{', '.join(l for l in sct_labels)}**: "
         else:
-            lines.append(f" - {title}. {compat_msg} [View pull request]({pr_url})\n")
-    return lines
+            labels_msg = ""
+
+        line = f" - {labels_msg}{title}. {compat_msg} [View pull request]({pr_url})\n"
+        lines.append((sorting_key, line))
+    return [line for (key, line) in sorted(lines)]
```

Comment:
The `sorting_key` trick is really interesting! From what I understand

- The `sorting_key` will look like e.g. `(['sct_deepseg'], 3981)` if the PR has an SCT function label, or `([], 3981)` if it doesn't.
- The `sorted` function sorts items from left to right, so:
    1. It will sort items by SCT function first.
    2. Then it will sort items by PR number.

The only downside to this is, `['sct_deepseg']` is ordered after `[]`, so non-labeled items come _before_ labeled items.

To get the `sct_` items to come first, we could do something like:

```diff
- sorting_key = (sct_labels, item['number'])
+ sorting_key = (sct_labels if sct_labels else ['x'], item['number'])
```

It's a little janky... having to use a phony list just to get the items to order after `['sct_']`? But it works!

Reply:
Since my suggestion makes things a little more complex...

To make the sorting a little more understandable, what if we inlined `sorting_key` (so that the sorting code is grouped together), then added a comment?

```python
        # Sorting precedence: 1. PR labels > 2. PR number > 3. Line contents
        # NB: CLI PRs (`sct_function`) are ordered before API PRs (denoted using 'x')
        lines.append((sct_labels if sct_labels else ['x'], item['number'], line))
    return [line for (labels, pr_number, line) in sorted(lines)]
```

Reply:
Done in [`dd8343b`](https://github.com/neuropoly/changelog/pull/33/commits/dd8343b9b8848b6f37cb20f5eb5d5a566bf202a1).